include/envoy/network/connection.h

```diff
@@ -116,6 +116,25 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
   virtual const std::string& remoteAddress() PURE;
 
   /**
+   * @return The port number used by the remote client.
+   */
+  virtual uint32_t remotePort() PURE;
```

Comment:
I'm not thrilled with how remoteAddress() is currently implemented, and will clean that up at some point, but currently remoteAddress already contains the port if it is a TCP connection. E.g., tcp://1.2.3.4:8080. There may be no remote port if this is a UDS connection also. I would just delete this new function, and get the port out of remoteAddress if it is a TCP connection via the existing utility functions. Similarly, I would just add destinationAddress(), which should look like tcp://1.2.3.4:8080, and remove destinationPort(), so that we can also handle UDS, UDP, etc. connections in the future. I understand the strings are hack, but we will fix that later.